lib/segment/src/vector_storage/chunked_vectors.rs

```diff
@@ -58,6 +58,25 @@ impl<T: Copy + Clone + Default> ChunkedVectors<T> {
         &chunk_data[idx..idx + self.dim]
     }
 
+    /// Duplicate of `get` with explicit bound check.
+    pub fn get_opt<TKey>(&self, key: TKey) -> Option<&[T]>
+    where
+        TKey: num_traits::cast::AsPrimitive<usize>,
+    {
+        if self.chunks.is_empty() {
+            return None;
+        }
+        let key: usize = key.as_();
+        let chunk_data = &self.chunks[key / self.chunk_capacity];
+        let idx = (key % self.chunk_capacity) * self.dim;
+        let range = idx..idx + self.dim;
+        if range.start < chunk_data.len() && range.end <= chunk_data.len() {
+            Some(&chunk_data[range])
+        } else {
+            None
+        }
```

Comment:
We can utilize `slice.get(0..4)` with a range here, which already returns an option.